library/cc/request_headers_builder.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+
+#include "headers_builder.h"
+#include "request_headers.h"
+#include "request_method.h"
+#include "retry_policy.h"
+#include "upstream_http_protocol.h"
+
+class RequestHeaders;
+
+class RequestHeadersBuilder : public HeadersBuilder {
+public:
+  RequestHeadersBuilder(RequestMethod::_RequestMethod request_method, const std::string& scheme,
+                        const std::string& authority, const std::string& path);
+
+  using HeadersBuilder::add;
```

Comment:
C++ question:

1. do I even need this, or will inheritance automatically allow subclasses to have the superclass's implementation?

2. i have a feeling that this will return a `HeadersBuilder&` instead of a `RequestHeadersBuilder&`, is that right?

Reply:
Subclasses will indeed inherit their superclasses' interface/implementations.